apcd-cms/src/client/src/components/Forms/Registrations/RegistrationForm.tsx

```diff
@@ -0,0 +1,504 @@
+        total_covered_lives: any;
+        claims_encounters_volume: any;
+        total_claims_value: any;
+    }[];
+    contacts: {
+        contact_type: string;
+        contact_name: string;
+        contact_phone: string;
+        contact_email: string;
+        contact_notifications: boolean;
+    }[];
+};
+  
+const initialValues: FormValues = {
+    on_behalf_of: "true",
+    reg_year: '',
+    type: "carrier",
+    business_name: '',
+    mailing_address: '',
+    city: '',
+    zip_code: '',
+    entities: [
+        {
+            entity_name: '',
+            fein: '',
+            license_number: '',
+            naic_company_code: '',
+            types_of_plans_commerical: false,
+            types_of_plans_medicare: false,
+            types_of_plans_medicaid: false,
+            types_of_files_eligibility_enrollment: true,
+            types_of_files_provider: false,
+            types_of_files_medical: false,
+            types_of_files_pharmacy: false,
+            types_of_files_dental: false,
+            total_covered_lives: '',
+            claims_encounters_volume: '',
+            total_claims_value: ''
+        },
+    ],
+    contacts: [
+        {
+            contact_type: '',
+            contact_name: '',
+            contact_phone: '',
+            contact_email: '',
+            contact_notifications: false
+        }
+    ]
+};
+
+
+export const RegistrationForm: React.FC = () => {
+    const { data, isLoading, isError } = useFormData({});
+    const [errorMessage, setErrorMessage] = useState('');
+    const [isSuccess, setIsSuccess] = useState<boolean>(false);
+    const [regId, setRegId] = useState(null);
+    
+    const handleSubmit = async (
+        values: FormValues,
+        { setSubmitting }: { setSubmitting: (isSubmitting: boolean) => void }
+    ) => {
+        setErrorMessage('');
+        const url = `register/request-to-submit/api/`;
+        try {
+            const response = await fetchUtil({
+                url,
+                method: `POST`,
+                body: values,
+            });
+            if (response.status == 'success') {
+                setIsSuccess(true);
+                setRegId(response.reg_id);
+            }
+        } catch (error: any) {
+            console.error('Error saving data:', error);
+            console.log(url);
+            if (error.response && error.response.data) {
+                setErrorMessage(
+                    error.response.data.message ||
+                    'An error occurred while saving the data. Please try again.'
+                );
+            } else {
+                setErrorMessage(
+                    'An error occurred while saving the data. Please try again.'
+                );
+            }
+        } finally {
+          setSubmitting(false);
+        }      
+    };
+
+    if (isLoading) {
+        <div className="loadingField">
+            <LoadingSpinner />
+        </div>
+    }
```

Comment:
```suggestion
    if (isLoading) {
    return (
        <div className="loadingField">
            <LoadingSpinner />
        </div>
        )
    }
```

Needs to be returned to render on load

Reply:
Good catch, thanks!